homeassistant/components/mopeka/__init__.py

```diff
@@ -22,11 +24,12 @@
 type MopekaConfigEntry = ConfigEntry[PassiveBluetoothProcessorCoordinator]
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: MopekaConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Mopeka BLE device from a config entry."""
     address = entry.unique_id
     assert address is not None
-    data = MopekaIOTBluetoothDeviceData()
+
+    data = MopekaIOTBluetoothDeviceData(MediumType(entry.data.get(DATA_MEDIUM_TYPE)))
```

Comment:
```suggestion
    data = MopekaIOTBluetoothDeviceData(MediumType(entry.data.get(CONF_MEDIUM_TYPE)))
```

Reply:
This needs a bit more work though since `entry.data.get(CONF_MEDIUM_TYPE)` will be `None` for existing entries so it should probably be
```python
medium_str: str | None = entry.data.get(CONF_MEDIUM_TYPE)
medium_type = MediumType(medium_str) if medium_str else None
```